src/pydvl/influence/inversion_methods.py

```diff
@@ -0,0 +1,84 @@
+"""
+Contains
+
+- batched conjugate gradient.
+- error bound for conjugate gradient.
+"""
+import logging
+from enum import Enum
+from typing import Callable, Tuple
+
+import numpy as np
+from numpy.typing import NDArray
+from scipy.sparse.linalg import LinearOperator, cg
+
+from ..utils import maybe_progress
+from .frameworks import TensorType
+
+__all__ = ["matrix_inversion_algorithm", "conjugate_gradient"]
+
+logger = logging.getLogger(__name__)
+
+
+class InversionMethod(str, Enum):
+    """
+    Different inversion methods types.
+    """
+
+    Direct = "direct"
+    Cg = "cg"
+
+
+def invert_matrix(
+    inversion_method: InversionMethod,
+    mvp: Callable[[TensorType], NDArray[np.float_]],
+    mvp_dimensions: Tuple[int, int],
+    b: NDArray[np.float_],
+    progress: bool = False,
+) -> NDArray[np.float_]:
```

Comment:
This should return TensorType as well, thus making `compute_influences` also consistent with types

Reply:
types should be fine now